test/parse.jl

```diff
@@ -1276,3 +1276,13 @@ end
 @test parse("(::A)") == Expr(Symbol("::"), :A)
 @test_throws ParseError parse("(::, 1)")
 @test_throws ParseError parse("(1, ::)")
+
+# issue #23014
+@test parse("Base.:==") == parse("Base.:(==)")
+try
+    parse(":=")
```

Comment:
this should have an error after it, otherwise if it fails to throw it'll silently succeed

or better, `@test_throws` returns the exception object I believe